components/automate-cli/pkg/verifyserver/server/api/v1/softwareversion_test.go

```diff
@@ -0,0 +1,185 @@
+package v1_test
+
+import (
+	"errors"
+	"io/ioutil"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/server"
+	v1 "github.com/chef/automate/components/automate-cli/pkg/verifyserver/server/api/v1"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/services/softwareversionservice"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/fiberutils"
+	"github.com/chef/automate/lib/logger"
+	"github.com/gofiber/fiber"
+	"github.com/stretchr/testify/assert"
+)
+
+func SetupMockSoftwareVersionService(response models.SoftwareVersionDetails, err error) softwareversionservice.ISoftwareVersionService {
+	return &softwareversionservice.MockSoftwareVersionService{
+		GetSoftwareDetailsFunc: func(string) (*models.SoftwareVersionDetails, error) {
+			return &response, err
+		},
+	}
+}
+
+func SetupHandler(sv softwareversionservice.ISoftwareVersionService) *fiber.App {
+	log, _ := logger.NewLogger("text", "debug")
+	fconf := &fiber.Settings{
+		ServerHeader: server.SERVICE,
+		ErrorHandler: fiberutils.CustomErrorHandler,
+	}
+	app := fiber.New(fconf)
+	handler := v1.NewHandler(log).AddSoftwareVersionService(sv)
+
+	vs := &server.VerifyServer{
+		Port:    server.DEFAULT_PORT,
+		Log:     log,
+		App:     app,
+		Handler: handler,
+	}
+	vs.SetupRoutes()
+	return vs.App
+}
+
+func TestSoftwareVersionAPI(t *testing.T) {
+	tests := []struct {
+		description   string
+		expectedCode  int
+		responseBody  models.SoftwareVersionDetails
+		expectedBody  string
+		expectedError error
+		query         string
+	}{
+		{
+			description:  "200:success software version check route when query is opensearch",
+			expectedCode: 200,
+			responseBody: models.SoftwareVersionDetails{
+				Passed: true,
+				Checks: []models.Checks{
+					{
+						Title:         "openssl availability",
+						Passed:        true,
+						SuccessMsg:    "openssl is available",
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         "mkdir availability",
+						Passed:        true,
+						SuccessMsg:    "mkdir is available",
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         "Linux Version Check",
+						Passed:        true,
+						SuccessMsg:    "Ubuntu version is 20.04",
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+				},
+			},
+			expectedBody:  `{"status":"SUCCESS","result":{"passed":true,"checks":[{"title":"openssl availability","passed":true,"success_msg":"openssl is available","error_msg":"","resolution_msg":""},{"title":"mkdir availability","passed":true,"success_msg":"mkdir is available","error_msg":"","resolution_msg":""},{"title":"Linux Version Check","passed":true,"success_msg":"Ubuntu version is 20.04","error_msg":"","resolution_msg":""}]}}`,
```

Comment:
You can use  &models.Checks{} and get success/ error message from variable( like you have done for **responseBody**) or create global variable for entire expectedBody